curl.opam

```diff
@@ -13,6 +13,7 @@ bug-reports: "https://github.com/ygrek/ocurl/issues"
 depends: [
   "dune" {>= "3.0"}
   "ocaml" {>= "4.03.0"}
```

Comment:
i think we can raise minimum ocaml version
one easy point of reference is debian oldstable - bullseye was initially released in 2021 and has OCaml 4.11.1 (from 2020), so lets jump to 4.11.1